src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -647,6 +721,103 @@ public static boolean isDST(String timezone) {
     return !zoneId.getRules().getTransitionRules().isEmpty();
   }
 
+  /**
+   * Read from `sun.util.calendar.ZoneInfo` via reflection to get timezone info.
+   * 
+   * @param tzId timezone id
+   * @return timezone info
+   */
+  private static TzInfoInJavaUtilForORC getInfoForUtilTZ(String tzId) {
+    TimeZone tz = TimeZone.getTimeZone(tzId);
+    if (!(tz instanceof ZoneInfo)) {
+      throw new UnsupportedOperationException("Unsupported timezone: " + tzId);
+    }
+    ZoneInfo zoneInfo = (ZoneInfo) tz;
+
+    // The constructor of TimeZoneInfoInJavaUtilPackage will extract and repack
+    // transitions info.
+    return new TzInfoInJavaUtilForORC(
+        (long[]) FieldUtils.readField(zoneInfo, "transitions"),
+        (int[]) FieldUtils.readField(zoneInfo, "offsets"),
+        (int) FieldUtils.readField(zoneInfo, "rawOffset"));
+  }
+
+  private static ColumnVector getTransitionsForUtilTZ(TzInfoInJavaUtilForORC info) {
+    try (HostColumnVector hcv = HostColumnVector.fromLongs(info.transitions)) {
+      return hcv.copyToDevice();
+    }
+  }
+
+  /**
+   * Get timezone info from `java.util.TimeZone`.
+   * 
+   * @param info timezone info
+   * @return a table on GPU containing timezone info from `java.util.TimeZone`
+   */
+  private static Table getTableForUtilTZ(TzInfoInJavaUtilForORC info) {
+    if (info.transitions == null) {
+      // fixed offset timezone
+      return null;
+    }
+    try (ColumnVector trans = getTransitionsForUtilTZ(info)) {
+      return new Table(trans);
+    } catch (Exception e) {
+      throw new IllegalStateException("get timezone info from java.util.TimeZone failed!", e);
+    }
+  }
+
+  /**
+   * Is Daylight Saving Time for the given timezone id
+   */
+  private static boolean isDaylightSavingTime(String timezoneId) {
```

Comment:
This isn't a check to see if the current time is within DST, right?  It is to check whether the specified timezone-id has any DST transitions at all?

I think `hasDaylightSavingTime` might be a more appropriate name.